subprojects/scala/src/test/groovy/org/gradle/api/tasks/ScalaRuntimeTest.groovy

```diff
@@ -21,18 +21,28 @@ import org.gradle.api.internal.file.collections.LazilyInitializedFileCollection
 import org.gradle.api.plugins.scala.ScalaBasePlugin
 import org.gradle.language.scala.internal.toolchain.DefaultScalaToolProvider
 import org.gradle.test.fixtures.AbstractProjectBuilderSpec
+import spock.lang.Unroll
 
 class ScalaRuntimeTest extends AbstractProjectBuilderSpec {
 
     def setup() {
         project.pluginManager.apply(ScalaBasePlugin)
     }
 
-    def "inferred Scala class path contains 'scala-compiler' repository dependency and 'compiler-bridge' matching 'scala-library' Jar found on class path"() {
+    @Unroll
+    def "inferred Scala class path contains 'scala-compiler' repository dependency and 'compiler-bridge' matching 'scala-library' Jar found on class path (default zinc: #defaultZinc)"() {
```

Comment:
I think it would make more sense to just do this as a separate test.  This is really testing two different things.

Reply:
I disagree. The test verifies we get the right libraries on the classpath. Since the `zincVersion` now impacts that, it makes sense to parameterize the test.
What would be the other test you are thinking about?

Reply:
Right, they have the same shape, but I think it's a smell to have a boolean parameter to switch between two different behaviors.  This sort of thing would be OK if we expected the exact same results regardless of the parameter, but in this case, the parameter changes the initial conditions (the `if (!defaultZinc)`) _and_ the expectations.  That basically means it's two tests trying to live in one method.  It's a minor thing, but the name of the test also suffers since you need to unpack how the boolean parameter affects the test vs just saying "with default Zinc version" and "with specified Zinc version" or whatever.  

I was thinking of something like this:
```
def "inferred Scala class path contains 'scala-compiler' repository dependency and 'compiler-bridge' matching 'scala-library' Jar found on class path"() {
        project.repositories {
            mavenCentral()
        }
        when:
        def classpath = project.scalaRuntime.inferScalaClasspath([new File("other.jar"), new File("scala-library-2.10.1.jar")])
        then:
        assertHasCorrectDependencies(classpath, DefaultScalaToolProvider.DEFAULT_ZINC_VERSION)
    }

    def "inferred Scala class path contains 'scala-compiler' repository dependency and 'compiler-bridge' matching 'scala-library' Jar found on class path with specified zinc version"() {
        project.repositories {
            mavenCentral()
        }
        def useZincVersion = "1.3.4"
        project.scala {
            zincVersion = useZincVersion
        }
        when:
        def classpath = project.scalaRuntime.inferScalaClasspath([new File("other.jar"), new File("scala-library-2.10.1.jar")])
        then:
        assertHasCorrectDependencies(classpath, useZincVersion)
    }

    private void assertHasCorrectDependencies(classpath, useZincVersion) {
        assert classpath instanceof LazilyInitializedFileCollection
        assert classpath.sourceCollections.size() == 1
        with(classpath.sourceCollections[0]) {
            assert it instanceof Configuration
            assert it.state == Configuration.State.UNRESOLVED
            assert it.dependencies.size() == 3
            assert it.dependencies.any { d ->
                d.group == "org.scala-lang" &&
                        d.name == "scala-compiler" &&
                        d.version == "2.10.1"
            }
            assert it.dependencies.any { d ->
                d.group == "org.scala-sbt" &&
                        d.name == "compiler-bridge_2.10" &&
                        d.version == useZincVersion
            }
            assert it.dependencies.any { d ->
                d.group == "org.scala-sbt" &&
                        d.name == "compiler-interface" &&
                        d.version == useZincVersion
            }
        }
    }
```

This version reduces the duplication between the tests by moving the common functionality into an assertion helper method.

I think it could also be OK to use a single test if the parameter was the version of Zinc to use and we just set `scala.zincVersion` to it.  This tests something slightly different, but I don't think it matters how the zincVersion is set when it comes to how `ScalaRuntime` works.

Reply:
I see your point, thanks for taking the time. Adapted the PR accordingly.